sklearn/utils/optimize.py

```diff
@@ -80,29 +139,36 @@ def newton_cg(func_grad_hess, func, grad, x0, args=(), eps=1e-4, tol=1e-4,
         ``max{|g_i | i = 1, ..., n} <= tol``
         where ``g_i`` is the i-th component of the gradient.
 
-    eps : float, optional
-        If fhess is approximated, use this value for the step size.
-
     maxiter : int
-        Number of iterations.
+        Number of Newton iterations.
+
+    maxinner : int
+        Number of CG iterations.
```

Comment:
For each dataset, I give the number of call to CG and the max/mean number of iteration inside CG.
If the dataset has more than 2 classes, we have several calls to `newton_cg`. I give the one with the hightest max and the one with the lowest max
```
boston:
37 CG call, n_loop: max = 7, mean = 2.3
45 CG call, n_loop: max = 18, mean = 4.7

iris:
11 CG call, n_loop: max = 3, mean = 1.8
11 CG call, n_loop: max = 4, mean = 2.2

digits:
19 CG call, n_loop: max = 11, mean = 3.7
30 CG call, n_loop: max = 65, mean = 13.3

rcv1.train:
20 CG call, n_loop: max = 10, mean = 5.9
```
I only reached `max > 100` with the following classification:
```
make_classification(n_samples=5000, n_features=2000, n_informative=200, 
                    n_classes=2, random_state=42)
```
```
24 CG call, n_loop: max = 109, mean = 34.6
time for fit: 18.8 sec
```
(or with `max_inner=100`, it calls one more CG)
```
25 CG call, n_loop: max = 100, mean = 36.8
time for fit: 20.0 sec
```